Dwolla.Client/HttpServices/DwollaHttpService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Dwolla.Client.HttpRequestServices;
+using Dwolla.Client.HttpServices.Architecture;
+using Dwolla.Client.Models;
+using Dwolla.Client.Models.Responses;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Dwolla.Client.HttpServices
+{
+	public class DwollaHttpService
+	{
+		private IDwollaClient _client;
+		public IDwollaClient Client
+		{
+			get => _client ?? DwollaConfiguration.Client;
+			set => _client = value;
+		}		
+		
+		private static TokenResponse _cachedToken;
+
+		private Task<string> getAccessTokenAsync;	
+
+		private async Task<string> GetAccessTokenAsync()
+		{
+			if (_cachedToken != null)//_cachedToken.ExpiresIn > now)
+			{ 
+				return _cachedToken.Token;
+			}
+
+			var response = await Authorization.GetToken();
+
+			if (response.Error != null)
+			{
+				throw new Exception(response.Error.Message);
+			}
+
+			_cachedToken = response.Content;
+
+			return _cachedToken.Token;
+		}
+
+		public DwollaHttpService()
+		{
+
+		}
+
+		public DwollaHttpService(Task<string> getAccessTokenAsync)
```

Comment:
You don't need to repeat the body of this constructor. You can invoke another constructor by doing something like `public DwollaHttpService(Task<string> getAccessTokenAsync) : this(null, getAccessTokenAsync) {}`